SlicerConnector/Startup.cs

```diff
@@ -225,55 +231,76 @@ private async Task HandleWebsocketConnection(HttpContext context, WebSocket webS
+                                WebSocketMessageType.Text, true, CancellationToken.None);
+                        };
+
+                        var localPath = Path.Combine(ModelDownloadPath, commands.File);
+                        if (await CheckForFileDifference(localPath, commands.File))
                         {
-                            var error = new ErrorMessage($"The requested file ({commands.File}) was not found on Octoprint").ToString();
-                            var errorBytes = Encoding.ASCII.GetBytes(error);
-                            await webSocket.SendAsync(new ArraySegment<byte>(errorBytes, 0, error.Length), WebSocketMessageType.Text, true, CancellationToken.None);
+                            var res = await os.FileOperations.DownloadFileAsync("local", commands.File, localPath);
+
+                            //downloading failed
+                            if (!res)
+                            {
+                                var error = new ErrorMessage(
+                                    $"The requested file ({commands.File}) was not found on Octoprint").ToString();
+                                var errorBytes = Encoding.ASCII.GetBytes(error);
+                                await webSocket.SendAsync(new ArraySegment<byte>(errorBytes, 0, error.Length),
+                                    WebSocketMessageType.Text, true, CancellationToken.None);
+                            }
+                            else
+                            {
+                                commands.File = localPath;
+                                await prusaSlicerBroker.SliceAsync(commands);
+                            }
+
                         }
+                        // use the local file on the disk
                         else
                         {
                             commands.File = localPath;
                             await prusaSlicerBroker.SliceAsync(commands);
                         }
-
-                    }
-                    // use the local file on the disk
-                    else
-                    {
-                        commands.File = localPath;
-                        await prusaSlicerBroker.SliceAsync(commands);
                     }
+
+                    result = await webSocket.ReceiveAsync(new ArraySegment<byte>(buffer), CancellationToken.None);
+                }
+                catch (JsonException)
+                {
+                    string msg =
+                        "The application was terminated because an incoming message from the web socket was not in proper Json format " +
+                        "Check that the websocket messages are in correct json format \n";
+
+                    Console.WriteLine(msg);
+                    await webSocket.CloseAsync(result.CloseStatus.Value, result.CloseStatusDescription, CancellationToken.None);
+                    throw;
                 }
```

Comment:
dont wrap the whole function in the try catch. move the try to the line where the string is parsed (and the exception is thrown).  Catch the exception immediately (not in the end of the function). Discard the incoming message if it is not in proper json format. DO NOT close the websocket connect. An Error message on the console is not necessary.